aten/src/ATen/core/Dict.h

```diff
@@ -0,0 +1,377 @@
+#pragma once
```

Comment:
There are two things in this file. A templated Dict that has few dependencies but is long. ~50 lines of code showing how to use IValue's inside of Dict.  Can we split them apart? It would help keeping the organization of ivalue.h understandable.

Reply:
These are not only used for `Dict<IValue, IValue>`, but for all dicts since even `Dict<int64_t, string>` represents its data as an `unordered_map<IValue, IValue>`. That's why these need to be defined before the `Dict` class.